templates/splunk-enterprise.template.yaml

```diff
@@ -478,6 +478,29 @@ Resources:
     CreationPolicy:
       ResourceSignal:
         Timeout: PT15M
+    Metadata:
+      AWS::CloudFormation::Init: 
+        config:
+          files: 
+            /tmp/user_data.sh:
+              source: !Join 
+                  - ''
+                  - - https://
+                    - !Ref 'QSS3BucketName'
+                    - .s3.amazonaws.com/
+                    - !Ref 'QSS3KeyPrefix'
+                    - scripts/user_data.sh
+              mode: '000755'
+              owner: root 
+              group: root
+              authentication: S3AccessCreds
+      AWS::CloudFormation::Authentication: 
+         S3AccessCreds:
+          type: S3
+          accessKeyId: !Ref 'CfnKeys'
+          secretKey: !GetAtt 'CfnKeys.SecretAccessKey'
+          buckets:
+            - !Ref 'QSS3BucketName'
```

Comment:
Why not use IAM role? Instead of creating user and using secret access key? That way you don't need to worry about rotating keys. Its much secure.